sktime/annotation/igts.py

```diff
@@ -0,0 +1,486 @@
+    ... means=[[0.0, 1.0], [11.0, 10.0], [5.0, 3.0], [2.0, 2.0]],
+    ... variances=0.5)
+    >>> X_scaled = MinMaxScaler(feature_range=(0, 1)).fit_transform(X)
+    >>> from sktime.annotation.igts import InformationGainSegmentation
+    >>> igts = InformationGainSegmentation(k_max=3, step=2)
+    >>> y = igts.fit_predict(X_scaled)
+
+    """
+
+    def __init__(
+        self,
+        k_max: int = 10,
+        step: int = 5,
+    ):
+        self._adaptee_class = IGTS
+        self._adaptee = self._adaptee_class(
+            k_max=k_max,
+            step=step,
+        )
+
+    def fit(self, X: npt.ArrayLike, y: npt.ArrayLike = None):
+        """Fit method for compatibility with sklearn-type estimator interface.
+
+        It sets the internal state of the estimator and returns the initialized
+        instance.
+
+        Parameters
+        ----------
+        X: array_like
+            2D `array_like` representing time series with sequence index along
+            the first dimension and value series as columns.
+
+        y: array_like
+            Placeholder for compatibility with sklearn-api, not used, default=None.
+        """
+        return self
+
+    def predict(self, X: npt.ArrayLike, y: npt.ArrayLike = None) -> npt.ArrayLike:
+        """Perform segmentation.
+
+        Parameters
+        ----------
+        X: array_like
+            2D `array_like` representing time series with sequence index along
+            the first dimension and value series as columns.
+
+        y: array_like
+            Placeholder for compatibility with sklearn-api, not used, default=None.
+
+        Returns
+        -------
+        y_pred : array_like
+            1D array with predicted segmentation of the same size as the first
+            dimension of X. The numerical values represent distinct segments
+            labels for each of the data points.
+        """
+        self.change_points_ = self._adaptee.find_change_points(X)
+        self.intermediate_results_ = self._adaptee.intermediate_results_
+        return self.to_clusters(self.change_points_)
+
+    def fit_predict(self, X: npt.ArrayLike, y: npt.ArrayLike = None) -> npt.ArrayLike:
+        """Perform segmentation.
+
+        A convenience method for compatibility with sklearn-like api.
+
+        Parameters
+        ----------
+        X: array_like
+            2D `array_like` representing time series with sequence index along
+            the first dimension and value series as columns.
+
+        y: array_like
+            Placeholder for compatibility with sklearn-api, not used, default=None.
+
+        Returns
+        -------
+        y_pred : array_like
+            1D array with predicted segmentation of the same size as the first
+            dimension of X. The numerical values represent distinct segments
+            labels for each of the data points.
+        """
+        return self.fit(X=X, y=y).predict(X=X, y=y)
```

Comment:
this should go and we can inherit from `from sklearn.base import BaseEstimator` which is implementing that?

Reply:
Right now `BaseEstimator` does not have this method https://github.com/sktime/sktime/blob/main/sktime/base/_base.py and it's required to pass the tests.